File: 4-medianTwoSortedArrays/Python3/medianTwoSortedArrays_test.py
import unittest
from medianTwoSortedArrays import Solution

class Test(unittest.TestCase):
    def test0(self):
        in1 = [0]
        in2 = [1,2]
        want = 1
        got = Solution().findMedianSortedArrays(in1, in2)
        print(f'Test {1} - got: {got} | want: {want}')
        self.assertEqual(got, want)
    def test1(self):
        in1 = [0]
        in2 = [1]
        want = 0.5
        got = Solution().findMedianSortedArrays(in1, in2)
        print(f'Test {1} - got: {got} | want: {want}')
        self.assertEqual(got, want)
    def test2(self):
        in1 = [0]
        in2 = [1,2]
        want = 1
        got = Solution().findMedianSortedArrays(in1, in2)
        print(f'Test {1} - got: {got} | want: {want}')
        self.assertEqual(got, want)	
    def test3(self):
        in1 = [0, 2]
        in2 = [1,2]
        want = 1.5
        got = Solution().findMedianSortedArrays(in1, in2)
        print(f'Test {1} - got: {got} | want: {want}')
        self.assertEqual(got, want)
    def test4(self):
        in1 = [-1,2]
        in2 = [1,-4]
        want = 0.0
        got = Solution().findMedianSortedArrays(in1, in2)
        print(f'Test {1} - got: {got} | want: {want}')
        self.assertEqual(got, want)

if __name__ == '__main__':
    unittest.main()

File: 4-medianTwoSortedArrays/JS/medianTwoSortedArrays.test.js
const findMedianSortedArrays = require("./medianTwoSortedArrays");

function expectations(n1, n2, want) {
  let got = findMedianSortedArrays(n1, n2);

  expect(got).toStrictEqual(want);
  expect(got).not.toBeLessThan(want);
  expect(got).not.toBeGreaterThan(want);
  expect(got).not.toBe(null);
}

test("empty", () => {
  let num = [0];
  let want = 0;

  expectations(num, num, want);
});

test("tests", () => {
  let n1 = [1, 2, 3];
  let n2 = [3, 4, 5];
  let want = 3.0;

  expectations(n1, n2, want);
});

test("tests", () => {
  let n1 = [];
  let n2 = [3, 4, 5];
  let want = 4.0;

  expectations(n1, n2, want);
});

test("tests", () => {
  let n1 = [1, 2];
  let n2 = [3, 4, 5];
  let want = 3.0;

  expectations(n1, n2, want);
});

test("tests", () => {
  let n1 = [1, 2, 2];
  let n2 = [3, 4, 5];
  let want = 2.5;

  expectations(n1, n2, want);
});
